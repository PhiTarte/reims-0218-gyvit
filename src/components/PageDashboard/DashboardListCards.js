import React from "react"
import { Container, Row, Col } from "reactstrap"
import PropTypes from "prop-types"

import DashboardCard from "./DashboardCard"
import { elements } from "../../links/elements"

const DashboardListCards = ({ mail, firstname }) => (
  <Container fluid>
    <Row>
      <Col md="12">
        <p
          style={{
            fontSize: "24px",
            height: "33px",
            marginTop: "5%",
            textAlign: "center"
          }}
        >
          Welcome to Gyvit, Romain.
        </p>

        <p
          style={{
            fontSize: "18px",
            height: "24px",
            textAlign: "center",
            color: "#000000",
            opacity: 0.45,
            marginBottom: "60px"
          }}
        >
          What would you like to do ?
        </p>
      </Col>
    </Row>
    <Row className="d-flex justify-content-around">
      {elements.map(element => <DashboardCard {...element} />)}
    </Row>
  </Container>
)

export default DashboardListCards
