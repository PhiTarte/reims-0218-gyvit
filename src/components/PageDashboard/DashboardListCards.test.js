import React from "react"
import ReactDOM from "react-dom"
import renderer from "react-test-renderer"

import DashboardListCards from "./DashboardListCards"

const elements = [
  {
    id: 0,
    image: "",
    titre: "Demande de documents",
    bio:
      "Faites le premier pas pour lancer votre stockage: ajoutez des objets physiques, des téléchargements numériques dont vous pouvez rêver",
    buttonText: "Créer la première requête",
    link: "lien0"
  },
  {
    id: 1,
    image: "",
    titre: "Partage de documents",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Share documents",
    link: "lien1"
  },
  {
    id: 2,
    image: "",
    titre: "Mes documents",
    bio:
      "Take the first step to launching your store. Add physical items, digital downloads you can dream up",
    buttonText: "Upload documents",
    link: "lien2"
  }
]

const name = "Kevin"

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<DashboardListCards elements={elements} name={name} />, div)
  ReactDOM.unmountComponentAtNode(div)
})
it("renders link html", () => {
  const tree = renderer
    .create(<DashboardListCards elements={elements} name={name} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
