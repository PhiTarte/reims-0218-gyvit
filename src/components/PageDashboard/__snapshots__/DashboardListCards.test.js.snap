// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders link html 1`] = `
<div
  className="container"
>
  <div
    className="App row"
  >
    <div
      className="col-sm-4 col-md-12"
    >
      <p
        style={
          Object {
            "fontSize": "24px",
            "height": "33px",
            "marginTop": "20%",
            "textAlign": "center",
          }
        }
      >
        Welcome to Gyvit, 
        Kevin
        .
      </p>
      <p
        style={
          Object {
            "color": "#000000",
            "fontSize": "18px",
            "height": "24px",
            "marginBottom": "60px",
            "opacity": 0.45,
            "textAlign": "center",
          }
        }
      >
        What would you like to do ?
      </p>
    </div>
  </div>
  <div
    className="d-flex justify-content-around row"
  >
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="card"
            style={
              Object {
                "height": "423px",
                "marginRight": "32px",
                "width": "288px",
              }
            }
          >
            <img
              alt="folder"
              className="card-img"
              src="/img/request.png"
              style={
                Object {
                  "height": "120px",
                  "marginLeft": "72px",
                  "marginRight": "64.8px",
                  "marginTop": "31px",
                  "width": "151px",
                }
              }
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": "0px",
                }
              }
            >
              <h5
                className="card-title"
                style={
                  Object {
                    "fontSize": "18px",
                    "height": "24px",
                    "marginLeft": "57px",
                    "marginRight": "56px",
                    "marginTop": "15px",
                    "paddingLeft": "0px",
                    "paddingRight": "0px",
                    "textAlign": "center",
                    "width": "175px",
                  }
                }
              >
                Documents requests
              </h5>
              <p
                className="card-text"
                style={
                  Object {
                    "fontSize": "15px",
                    "height": "96px",
                    "lineHeight": "24px",
                    "marginBottom": "50px",
                    "marginLeft": "30px",
                    "marginRight": "30px",
                    "marginTop": "13px",
                    "opacity": 0.45,
                    "textAlign": "center",
                    "width": "228px",
                  }
                }
              >
                Take the first step to launching your store. Add physical items, digital downloads you can dream up.
              </p>
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                src="lien0"
                style={
                  Object {
                    "borderRadius": "2px",
                    "fontSize": "16px",
                    "height": "44px",
                    "marginBottom": "30px",
                    "marginLeft": "59px ",
                    "marginRight": "59px",
                    "padding": "0px",
                    "width": "170px",
                  }
                }
                type={undefined}
              >
                Create first request
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="card"
            style={
              Object {
                "height": "423px",
                "marginRight": "32px",
                "width": "288px",
              }
            }
          >
            <img
              alt="folder"
              className="card-img"
              src="/img/share.png"
              style={
                Object {
                  "height": "120px",
                  "marginLeft": "72px",
                  "marginRight": "64.8px",
                  "marginTop": "31px",
                  "width": "151px",
                }
              }
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": "0px",
                }
              }
            >
              <h5
                className="card-title"
                style={
                  Object {
                    "fontSize": "18px",
                    "height": "24px",
                    "marginLeft": "57px",
                    "marginRight": "56px",
                    "marginTop": "15px",
                    "paddingLeft": "0px",
                    "paddingRight": "0px",
                    "textAlign": "center",
                    "width": "175px",
                  }
                }
              >
                Share documents
              </h5>
              <p
                className="card-text"
                style={
                  Object {
                    "fontSize": "15px",
                    "height": "96px",
                    "lineHeight": "24px",
                    "marginBottom": "50px",
                    "marginLeft": "30px",
                    "marginRight": "30px",
                    "marginTop": "13px",
                    "opacity": 0.45,
                    "textAlign": "center",
                    "width": "228px",
                  }
                }
              >
                Take the first step to launching your store. Add physical items, digital downloads you can dream up.
              </p>
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                src="lien1"
                style={
                  Object {
                    "borderRadius": "2px",
                    "fontSize": "16px",
                    "height": "44px",
                    "marginBottom": "30px",
                    "marginLeft": "59px ",
                    "marginRight": "59px",
                    "padding": "0px",
                    "width": "170px",
                  }
                }
                type={undefined}
              >
                Share documents
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="card"
            style={
              Object {
                "height": "423px",
                "marginRight": "32px",
                "width": "288px",
              }
            }
          >
            <img
              alt="folder"
              className="card-img"
              src="/img/store.png"
              style={
                Object {
                  "height": "120px",
                  "marginLeft": "72px",
                  "marginRight": "64.8px",
                  "marginTop": "31px",
                  "width": "151px",
                }
              }
            />
            <div
              className="card-body"
              style={
                Object {
                  "padding": "0px",
                }
              }
            >
              <h5
                className="card-title"
                style={
                  Object {
                    "fontSize": "18px",
                    "height": "24px",
                    "marginLeft": "57px",
                    "marginRight": "56px",
                    "marginTop": "15px",
                    "paddingLeft": "0px",
                    "paddingRight": "0px",
                    "textAlign": "center",
                    "width": "175px",
                  }
                }
              >
                Store files
              </h5>
              <p
                className="card-text"
                style={
                  Object {
                    "fontSize": "15px",
                    "height": "96px",
                    "lineHeight": "24px",
                    "marginBottom": "50px",
                    "marginLeft": "30px",
                    "marginRight": "30px",
                    "marginTop": "13px",
                    "opacity": 0.45,
                    "textAlign": "center",
                    "width": "228px",
                  }
                }
              >
                Take the first step to launching your store. Add physical items, digital downloads you can dream up.
              </p>
              <button
                className="btn btn-secondary"
                onClick={[Function]}
                src="lien2"
                style={
                  Object {
                    "borderRadius": "2px",
                    "fontSize": "16px",
                    "height": "44px",
                    "marginBottom": "30px",
                    "marginLeft": "59px ",
                    "marginRight": "59px",
                    "padding": "0px",
                    "width": "170px",
                  }
                }
                type={undefined}
              >
                Upload documents
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
